sentry/src/routes/campaign.rs

```diff
@@ -25,32 +68,197 @@ pub async fn create_campaign<A: Adapter>(
+                    ))
+                }
+                _ => Err(error_response),
+            }
+        }
+        Ok(false) => Err(ResponseError::BadRequest("Encountered error while creating Campaign; please try again".to_string())),
+        _ => Ok(()),
+    }?;
 
     Ok(success_response(serde_json::to_string(&campaign)?))
 }
 
+
+// tested
+fn get_total_remaining_for_channel(accounting_spent: &UnifiedNum, latest_spendable: &Spendable) -> Option<UnifiedNum> {
+    let total_deposited = latest_spendable.deposit.total;
+
+    let total_remaining = total_deposited.checked_sub(&accounting_spent);
+    total_remaining
+}
+
+pub mod update_campaign {
+    use super::*;
+    use lazy_static::lazy_static;
+
+    lazy_static!{
+        pub static ref CAMPAIGN_REMAINING_KEY: &'static str = "campaignRemaining";
+    }
+
+    pub async fn increase_remaining_for_campaign(redis: &MultiplexedConnection, id: CampaignId, amount: UnifiedNum) -> Result<bool, CampaignError> {
+        let key = format!("{}:{}", *CAMPAIGN_REMAINING_KEY, id);
+        redis::cmd("INCRBY")
+            .arg(&key)
+            .arg(amount.to_u64())
+            .query_async(&mut redis.clone())
+            .await?;
+        Ok(true)
+    }
+
+    pub async fn decrease_remaining_for_campaign(redis: &MultiplexedConnection, id: CampaignId, amount: UnifiedNum) -> Result<bool, CampaignError> {
+        let key = format!("{}:{}", *CAMPAIGN_REMAINING_KEY, id);
+        redis::cmd("DECRBY")
+            .arg(&key)
+            .arg(amount.to_u64())
+            .query_async(&mut redis.clone())
+            .await?;
+        Ok(true)
+    }
+
+    pub async fn handle_route<A: Adapter>(
+        req: Request<Body>,
+        app: &Application<A>,
+    ) -> Result<Response<Body>, ResponseError> {
+        let campaign = req.extensions().get::<Campaign>().expect("We must have a campaign in extensions");
+
+        let modified_campaign = ModifyCampaign::from_campaign(campaign.clone());
+
+        // modify Campaign
+        modify_campaign(&app.pool, &campaign, &modified_campaign, &app.redis).await.map_err(|_| ResponseError::BadRequest("Failed to update campaign".to_string()))?;
+
+        Ok(success_response(serde_json::to_string(&campaign)?))
+    }
+
+    pub async fn modify_campaign(pool: &DbPool, campaign: &Campaign, modified_campaign: &ModifyCampaign, redis: &MultiplexedConnection) -> Result<Campaign, CampaignError> {
+        // *NOTE*: When updating campaigns make sure sum(campaigns.map(getRemaining)) <= totalDepoisted - totalspent
+        // !WARNING!: totalSpent != sum(campaign.map(c => c.spending)) therefore we must always calculate remaining funds based on total_deposit - lastApprovedNewState.spenders[user]
+        // *NOTE*: To close a campaign set campaignBudget to campaignSpent so that spendable == 0
+
+        let new_budget = modified_campaign.budget.ok_or(CampaignError::FailedUpdate("Couldn't get new budget".to_string()))?;
```

Comment:
It's not mandatory to update the budget when you send a Modify Campaign request. This should **not** fail and instead handle the new Budget if it's `Some`
- [ ] Handle budget only if it's passed through the request